### PR TITLE
Refactor: Move `pseudo::set_pseudo` to `Pseudopot::complete_default`.

### PR DESCRIPTION
**Pull request by sunliang98**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Linked Issue
Fix #...

### Unit Tests and/or Case Tests for my changes
- Corresponding unit tests are updated.

### What's changed?
- Move `pseudo::set_pseudo` to `Pseudopot::complete_default`, so that `pseudo` only takes charge of holding the data, which is filled by `Pseudopot`.

### Any changes of core modules? (ignore if not applicable)
- Example: I have added a new virtual function in the esolver base class in order to ...

### CODE REVIEW

File: source/module_cell/read_pp.cpp [hunk 1]
```diff
@@ -487,3 +487,181 @@ void Pseudopot_upf::setqfnew(const int& nqf,
...
+	this->complete_default_atom(pp);
+	this->complete_default_vl(pp);
+
+	if (pp.nbeta == 0) {
+		return;
+	}
+	
+	if (pp.lll == nullptr)
+	{
+		pp.lll = new int[pp.nbeta];
```
Comment (mohanchen) [inline source/module_cell/read_pp.cpp]: Is it possible to use std::vector to replace new and delete?

File: source/module_cell/read_pp.cpp [hunk 2]
```diff
@@ -487,3 +487,181 @@ void Pseudopot_upf::setqfnew(const int& nqf,
...
+void Pseudopot_upf::complete_default(Atom_pseudo& pp)
+{
+    ModuleBase::TITLE("Pseudopot_upf", "complete_default");
+
+    // call subroutines
+    this->complete_default_h(pp);
+	this->complete_default_atom(pp);
+	this->complete_default_vl(pp);
+
+	if (pp.nbeta == 0) {
```
Comment (mohanchen) [inline source/module_cell/read_pp.cpp]: It's a bit dangerous that this file contains more than 500 lines, is it possible to split it into more files?

File: source/module_cell/read_pp.cpp (hunk 1 above)
Comment (sunliang98) [reply source/module_cell/read_pp.cpp]: Yes, if needed, I will replace all `new`/`delete` in `pseudo` class with `vector` in next pr.

File: source/module_cell/read_pp.cpp (hunk 2 above)
Comment (sunliang98) [reply source/module_cell/read_pp.cpp]: I have moved the functions corresponding to `complete_default` to a new file `read_pp_complete.cpp`.

**Review by mohanchen — APPROVED** [review]

**Pull request merged**